### PR TITLE
fix: correctly await types in serve

### PR DESCRIPTION
**Pull request by Caele**
## Motivation

Fixes so that serve awaits added types before running.

## Requirements checklist

- [ ] Api specification
  - [ ] Ran `yarn spec`
    - [ ] No changes **_OR_** API changes has been formally approved
- [ ] Unit/Component test coverage
- [ ] Correct PR title for the changes (fix, chore, feat)

When build and tests have passed:

- [ ] Add code reviewers, for example @qlik-oss/nebula-core

### CODE REVIEW

File: commands/serve/web/components/Visualize/Visualize.jsx [hunk 1]
```diff
@@ -97,11 +98,16 @@ export default function Visualize() {
...
+      if (info) {
+        await initiateWatch(info);
+        setInit(true);
+      }
+    })();
   }, [info]);
 
   useEffect(() => {
-    if (waiting) return;
+    if (waiting || !initialized) return;
```
Comment (linhnguyen-qlik) [inline commands/serve/web/components/Visualize/Visualize.jsx:110]: Do we need to add `initialized` to dependency list of the useEffect?

**Review by linhnguyen-qlik — APPROVED** [review]
Nice, thank you for fixing this :). LGTM I have only a small question.

File: commands/serve/web/components/Visualize/Visualize.jsx (hunk 1 above)
Comment (Caele) [reply commands/serve/web/components/Visualize/Visualize.jsx:110]: we do yes, good catch

File: commands/serve/web/hot.js [hunk 2]
```diff
@@ -62,4 +55,13 @@ export default function initiateWatch(info) {
...
   update();
+  if (info.types) {
+    const proms = info.types.map((t) => {
+      return getModule(t.name, t.url).then((mo) => {
+        window[t.name] = mo;
+      });
+    });
+    return Promise.all(proms);
+  }
+  return Promise.resolve();
```
Comment (T-Wizard) [inline commands/serve/web/hot.js:66]: [optional]
alternative you could set window[t.name] to a promise
```
window[t.name] = getModule(t.name, t.url);
```
then you would not need to wait here
and I think that would be closer to how it work in a real mashup

**Pull request merged**